lib/foreman_tasks/tasks/export_tasks.rake

```diff
@@ -218,35 +231,37 @@ namespace :foreman_tasks do
       end
     end
 
+    filter = ENV['FILTER'] || "result != success && started_at > \"#{60.days.ago.to_s(:db)}\""
+    export_filename = ENV['FILE'] || "/tmp/task-export-#{DateTime.now.to_i}"
+    format =  ENV['FORMAT'] || 'html'
 
-    if ENV['tasks'] == 'all'
-      tasks = ForemanTasks::Task
-    else
-      tasks = ForemanTasks::Task.where("result != 'success'")
-    end
-
-    days = ENV['days'].try(:to_i) || 60
-    tasks = tasks.where('started_at > ?', days.days.ago)
-    export_filename = ENV['export'] || "/tmp/task-export-#{DateTime.now.to_i}.tar.gz"
+    tasks = ForemanTasks::Task.search_for(filter)
+    export_filename += (format == 'csv' ? '.csv' : '.tar.gz')
 
-    puts _("Gathering last #{days} days of tasks.")
-    Dir.mktmpdir('task-export') do |tmp_dir|
-      PageHelper.copy_assets(tmp_dir)
+    puts _("Gathering #{tasks.count} tasks.")
+    if format == 'html'
+      Dir.mktmpdir('task-export') do |tmp_dir|
+        PageHelper.copy_assets(tmp_dir)
 
 
-      renderer = TaskRender.new
-      count = 1
-      total = tasks.count
+        renderer = TaskRender.new
+        total = tasks.count
 
-      tasks.all.each do |task|
-        File.open(File.join(tmp_dir, "#{task.id}.html"), 'w') {|file| file.write(PageHelper.pagify(renderer.render_task(task)))}
-        puts "#{count}/#{total}"
-        count += 1
-      end
+        tasks.each_with_index do |task, count|
+          File.open(File.join(tmp_dir, "#{task.id}.html"), 'w') {|file| file.write(PageHelper.pagify(renderer.render_task(task)))}
+          puts "#{count + 1}/#{total}"
+          count += 1
+        end
 
-      File.open(File.join(tmp_dir, "index.html"), 'w') {|file| file.write(PageHelper.pagify(PageHelper.generate_index(tasks)))}
+        File.open(File.join(tmp_dir, "index.html"), 'w') {|file| file.write(PageHelper.pagify(PageHelper.generate_index(tasks)))}
 
-      sh("tar cvzf #{export_filename} #{tmp_dir} > /dev/null")
+        sh("tar cvzf #{export_filename} #{tmp_dir} > /dev/null")
+      end
+    elsif format == 'csv'
+      CSV.open(export_filename, 'wb') do |csv|
+        csv << ['id', 'state', 'type', 'label']
```

Comment:
`result`, `started_at`, `ended_at` and `parent_task_id` are pretty useful data to gather as well.